### ISSUE TITLE
Instructions for gaw integration

### ISSUE THREAD

**Issue by bluecmd** [StefanSchippers/xschem#3]
Hi @StefanSchippers !

Thanks for a tool that looks really cool!
I found out about the gaw integration from [your video](https://xschem.sourceforge.io/stefan/xschem_man/video_tutorials/probe_to_gaw.mp4). Thanks for making that!

I'm using `xschem xschem_library/examples/0_examples_top.sch` and located myself inside the `rlc` part using `e`.
I have set up gaw to listen on port 2020.

When I do Netlist, Simulate, and then Waves to launch gaw everything works fine - gaw pops up and shows me the `rlc.raw` table (note! not `0_examples_top.raw`).

When I then try to Alt-G a wire I get the error in the terminal `gaw -> 0_examples_top.raw table not defined`. Which is correct, as the current table is `rlc.raw`.

I have also had issues where xschem "forgets" to add a wave. E.g. if you try to send a net to gaw without any panel selected, gaw will complain. If you then add a panel and try to re-send it, it appears xschem doesn't do anything and gaw doesn't display a new waveform. Maybe xschem thinks it successfully sent the net?

If I press Simulation -> Send highlighted nets to gaw, that makes the net pop up again, so at least there's a workaround.

In summary; it's a really cool feature with a lot of potential, but it's a bit easy to cut oneself on right now. Documentation would be good on how to debug it and how it is supposed to work.

Thanks!

**Comment by StefanSchippers** [comment]
Xschem handles simulation of hierarchical schematics. As such top level nodes just are referenced by their names, like v(a), v(b), v(enable). If you descend into a subcircuit (say instance name x1) the nodes therein are named v(x1.c), v(x1.out), and so on. Xschem keeps track of the hierarchy path from the top.
If you simulate rlc.sch you must open (as top level schematic, File->Open) rlc.sch, since this is the top level from where the netlist is created.
If you open the parent 0_examples_top.sch and descend into rlc to do the simulation the nodes sent to gaw will be v(x0.a), (x0.c) instead of just v(a), v(c).
Open as a top level the schematic you want to simulate and start gaw from there. After that you can navigate down into the hierarchy (if there are more levels) and probe lower level nodes to gaw.

>
> I have also had issues where xschem "forgets" to add a wave. E.g. if you try to send a net to gaw without any panel selected, gaw will complain. If you then add a panel and try to re-send it, it appears xschem doesn't do anything and gaw doesn't display a new waveform. Maybe xschem thinks it successfully sent the net?
>
> If I press Simulation -> Send highlighted nets to gaw, that makes the net pop up again, so at least there's a workaround.

I can not replicate this behaviour. Are panels selectable? If i fire gaw (from xschem Wave button) and do nothing in its window i can just send a net probe to it.

for the xschem part: Alt-g  highlights the net and sends the net to gaw (with the same color, this was one of my patches to gaw, its original socket command set did not allow to specify wave color). However if net in xschem is already colored doing Alt-g again does nothing. Clear xschem highlights (Shift-K) so you can send the net to gaw again. As you discovered the Simulation->Send... to gaw" always works with already highlighted nets.
